homeassistant/components/climate/mysensors.py

```diff
@@ -0,0 +1,238 @@
+"""
+mysensors platform that offers a Climate(MySensors-HVAC) component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/#climate
+
+Curently supported operations:
+
+set_temperature()    --> V_HVAC_SETPOINT_COOL, V_HVAC_SETPOINT_HEAT
+set_operation_mode() --> V_HVAC_FLOW_STATE
+set_fan_mode()       --> V_HVAC_SPEED
+
+Pending:
+humidity, away_mode, aux_heat, swing_mode
+
+home-assistant/components/climate/mysensors.py
+"""
+import logging
+
+from homeassistant.components import mysensors
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT, ATTR_TEMPERATURE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the mysensors climate."""
+    if discovery_info is None:
+        return
+    for gateway in mysensors.GATEWAYS.values():
+        pres = gateway.const.Presentation
+        set_req = gateway.const.SetReq
+        map_sv_types = {
+            pres.S_HVAC: [set_req.V_HVAC_SETPOINT_HEAT,
+                          set_req.V_HVAC_SETPOINT_COOL,
+                          set_req.V_HVAC_FLOW_STATE,
+                          set_req.V_HVAC_FLOW_MODE, set_req.V_HVAC_SPEED],
+        }
+        device_class_map = {
+            pres.S_HVAC: MySensorsHVAC,
+        }
+        devices = {}
+        gateway.platform_callbacks.append(mysensors.pf_callback_factory(
+            map_sv_types, devices, add_devices, device_class_map))
+
+
+# pylint: disable=too-many-arguments, too-many-public-methods
+# pylint: disable=too-many-instance-attributes
+class MySensorsHVAC(mysensors.MySensorsDeviceEntity, ClimateDevice):
+    """Representation of a MySensorsHVAC hvac."""
+
+    def __init__(self, *args):
+        """Setup instance attributes."""
+        mysensors.MySensorsDeviceEntity.__init__(self, *args)
+        self._state = None
+        # Default Target Temperature set to a comfortable value of 24 degrees
+        self._target_temperature = 24.0
+        self._target_humidity = None
+        self._away = None
+        self._current_temperature = None
+        self._current_humidity = None
+        self._current_fan_mode = "Auto"
+        self._current_operation = "Off"
+        self._aux = None
+        self._current_swing_mode = "Off"
+        self._fan_list = ["Auto", "Min", "Normal", "Max"]
+        self._operation_list = ["CoolOn", "Off"]
```

Comment:
You can actually import all four states from the climate component, as the climate component imports `STATE_OFF`.